src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/client/ServiceResponseParser.java

```diff
@@ -25,7 +28,11 @@ public void tryParseResponseBodyAndThrow(HttpStatusCodeException exception) thro
                 ClientServiceErrorResponse.class
             );
 
-            throw new InvalidCaseDataException(exception, errorResponse);
+            if (exception.getStatusCode().equals(BAD_REQUEST)) {
+                throw new InvalidCaseDataException(exception, errorResponse);
+            } else if (exception.getStatusCode().equals(UNPROCESSABLE_ENTITY)) {
+                throw new UnprocessableEntityException(exception, errorResponse);
+            }
```

Comment:
I've just realised that `HttpStatusCodeException` is an abstract class, whose concrete subclasses represent specific HTTP status codes. Given that, we can completely get rid of this parser and both exceptions. Instead, we can catch appropriate exceptions at service level. It will make it much simpler.

Reply:
I can make that change if you'd like.